python/tests/tornado/test_hyperbahn.py

```diff
@@ -57,8 +58,21 @@ def test_request():
 
 
 @pytest.mark.gen_test
-def test_register():
+def test_advertise(tchannel_server):
+    endpoint = b'ad'
+
+    f = Future()
+
+    @tornado.gen.coroutine
+    def execution(request, response):
+        body = yield request.get_body()
+        f.set_result(body)
+
+    tchannel_server.expect_call(endpoint, 'json').execute = execution
     channel = TChannel(name='test')
+    hyperbahn.advertise(channel,
+                        'test', ['localhost:'+str(tchannel_server.port)]
+                        )
 
-    with pytest.raises(ConnectionClosedError):
-        yield hyperbahn.advertise(channel, 'foo', ['127.0.0.1:23000'])
+    result = yield f
```

Comment:
The future is a nice idea but if the call never goes through, it'll freeze forever/until the pytest-tornado async timeout kicks in.

Also, you want to yield advertise.